lib/livebook_web/live/session_live.ex

```diff
@@ -1425,6 +1494,31 @@ defmodule LivebookWeb.SessionLive do
     )
   end
 
+  defp prune_outputs_in_data(data) do
+    %{
+      data
+      | notebook:
+          Notebook.update_cells(data.notebook, fn
+            %Cell.Elixir{} = cell ->
+              %{cell | outputs: Enum.map(cell.outputs, &prune_output/1)}
+
+            cell ->
+              cell
+          end)
+    }
+  end
+
+  defp prune_output({:stdout, _}), do: {:stdout, :__pruned__}
+  defp prune_output({:text, _}), do: {:text, :__pruned__}
+  defp prune_output({:image, _, _}), do: {:image, :__pruned__, :__pruned__}
+  defp prune_output({:markdown, _}), do: {:markdown, :__pruned__}
+
+  defp prune_output({:frame, outputs, info}) do
+    {:frame, Enum.map(outputs, &prune_output/1), info}
+  end
+
+  defp prune_output(output), do: output
```

Comment:
@josevalim we need to keep basic output structure.

For example we need to keep inputs inside frame, because we need to know when to garbage collect input values that we keep. Other than that having the output type helps in keeping the proper output borders.

Similarly, for frame we don't want `phx-update="append"`, because there may be inputs and they are fine to re-render.

So basically we prune the outputs from `data` on the initial render and then whenever a new output is added. Note that the outputs are in `data_view` for things to render, and next time we compute `data_view` the pruning propagates there automatically.

For frame and stdout we send directly to the respective components.